app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/DLCStatus.scala

```diff
@@ -0,0 +1,295 @@
+  }
+
+  case class Signed(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer,
+      accept: DLCAccept,
+      sign: DLCSign)
+      extends AcceptedDLCStatus {
+
+    def toBroadcasted(fundingTx: Transaction): Broadcasted = {
+      Broadcasted(eventId, isInitiator, offer, accept, sign, fundingTx)
+    }
+
+    def toConfirmed(fundingTx: Transaction): Confirmed = {
+      toBroadcasted(fundingTx).toConfirmed
+    }
+  }
+
+  case class Broadcasted(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer,
+      accept: DLCAccept,
+      sign: DLCSign,
+      fundingTx: Transaction)
+      extends AcceptedDLCStatus {
+
+    def toConfirmed: Confirmed = {
+      Confirmed(eventId, isInitiator, offer, accept, sign, fundingTx)
+    }
+  }
+
+  case class Confirmed(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer,
+      accept: DLCAccept,
+      sign: DLCSign,
+      fundingTx: Transaction)
+      extends AcceptedDLCStatus {
+
+    def toCloseOffered(closeSig: DLCMutualCloseSig): CloseOffered = {
+      CloseOffered(eventId,
+                   isInitiator,
+                   offer,
+                   accept,
+                   sign,
+                   fundingTx,
+                   closeSig)
+    }
+
+    def toClaiming(
+        oracleSig: SchnorrDigitalSignature,
+        cet: Transaction): Claiming = {
+      Claiming(eventId,
+               isInitiator,
+               offer,
+               accept,
+               sign,
+               fundingTx,
+               oracleSig,
+               cet)
+    }
+
+    def toRemoteClaiming(cet: Transaction): RemoteClaiming = {
+      RemoteClaiming(eventId, isInitiator, offer, accept, sign, fundingTx, cet)
+    }
+  }
+
+  case class CloseOffered(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer,
+      accept: DLCAccept,
+      sign: DLCSign,
+      fundingTx: Transaction,
+      closeSig: DLCMutualCloseSig)
+      extends AcceptedDLCStatus {
+
+    def toClosed(closeTx: Transaction): Closed = {
+      Closed(eventId,
+             isInitiator,
+             offer,
+             accept,
+             sign,
+             fundingTx,
+             closeSig,
+             closeTx)
+    }
+
+    def toRefunded(refundTx: Transaction): Refunded = {
+      Refunded(eventId, isInitiator, offer, accept, sign, fundingTx, refundTx)
+    }
+  }
+
+  case class Closed(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer,
+      accept: DLCAccept,
+      sign: DLCSign,
+      fundingTx: Transaction,
+      closeSig: DLCMutualCloseSig,
+      closeTx: Transaction)
+      extends AcceptedDLCStatus
+
+  case class Claiming(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer,
+      accept: DLCAccept,
+      sign: DLCSign,
+      fundingTx: Transaction,
+      oracleSig: SchnorrDigitalSignature,
+      cet: Transaction)
```

Comment:
Isn't this required to be a `WitnessTransaction`? Let's tighten the type if we can. The same applies to other CETs here I think. I think the funding transaction can be non-witness tx though right?